src/sql/workbench/contrib/scripting/browser/scriptingActions.ts

```diff
@@ -311,34 +311,18 @@ CommandsRegistry.registerCommand({
 CommandsRegistry.registerCommand({
 	id: OE_REFRESH_COMMAND_ID,
 	handler: async (accessor, args: ObjectExplorerActionsContext): Promise<void> => {
-		const connectionManagementService = accessor.get(IConnectionManagementService);
-		const capabilitiesService = accessor.get(ICapabilitiesService);
 		const objectExplorerService = accessor.get(IObjectExplorerService);
 		const logService = accessor.get(ILogService);
 		const notificationService = accessor.get(INotificationService);
-		const connection = new ConnectionProfile(capabilitiesService, args.connectionProfile);
-		if (connectionManagementService.isConnected(undefined, connection)) {
-			let treeNode = await getTreeNode(args, objectExplorerService);
-			if (!treeNode) {
-				await objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile());
-				treeNode = objectExplorerService.getObjectExplorerNode(connection);
-			}
-			if (treeNode) {
-				const tree = objectExplorerService.getServerTreeView().tree;
-				try {
-					await tree.collapse(treeNode);
-					await objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode);
-					await tree.refresh(treeNode);
-					await tree.expand(treeNode);
-				} catch (err) {
-					// Display message to the user but also log the entire error to the console for the stack trace
-					notificationService.error(localize('refreshError', "An error occurred refreshing node '{0}': {1}", args.nodeInfo.label, getErrorMessage(err)));
-					logService.error(err);
-				}
-
-			} else {
-				logService.error(`Could not find tree node for node ${args.nodeInfo.label}`);
-			}
+		const treeNode = await getTreeNode(args, objectExplorerService);
+		const tree = objectExplorerService.getServerTreeView().tree;
+		try {
+			await objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode);
+			await tree.refresh(treeNode);
```

Comment:
and the collapse and expand are no longer needed either?

Reply:
That was removed on purpose - not part of a bug fix but rather just what I saw as bad behavior while fixing this. Collapsing and expanding the node caused a brief "flicker" in the tree as it repainted everything. And as far as I can tell it doesn't seem to benefit anything to collapse and re-expand the node.

This can be done in a later PR though if we think it's too risky though.